src/Microsoft.AspNetCore.Routing/Builder/MapEndpointEndpointDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class MapEndpointEndpointDataSourceBuilderExtensions
+    {
+        public static EndpointBuilder MapEndpoint(
+            this EndpointDataSourceBuilder builder,
+            Func<RequestDelegate, RequestDelegate> invoker,
+            string name,
```

Comment:
this should be `displayName`, and come after route pattern (like matcher endpoint). It would be good to have these args follow the same order as `MatcherEndpoint` if possible.

Reply:
For these methods I followed `MapRouteRouteBuilderExtensions.MapRoute` which has name before template.

Why is it important to follow `MatcherEndpoint`?

Reply:
Oh is this route name or display name? I'd really, really like to keep old concepts like RVA and route name out of these apis

Reply:
It is the display name. I'll rename it.

Reply:
This should be `MatcherEndpointBuilder`.